src/node_builtins.cc

```diff
@@ -219,6 +239,32 @@ MaybeLocal<String> BuiltinLoader::LoadBuiltinSource(Isolate* isolate,
 #endif  // NODE_BUILTIN_MODULES_PATH
 }
 
+static std::unique_ptr<icu::UnicodeString> save_source_utf16 = nullptr;
+void BuiltinLoader::AddExternalizedBuiltin(const char* id,
+                                           const char* filename) {
+  std::string source;
+  int r = ReadFileSync(&source, filename);
+  if (r != 0) {
+    fprintf(stderr,
+            "Cannot load externalized builtin: \"%s:%s\".\n",
+            id,
+            filename);
+    ABORT();
+    return;
+  }
+
+  icu::UnicodeString utf16 = icu::UnicodeString::fromUTF8(
+      icu::StringPiece(source.data(), source.length()));
+  auto source_utf16 = std::make_unique<icu::UnicodeString>(utf16);
```

Comment:
This probably needs some sort of `#ifdef` guard for `--without-intl`?

Reply:
@addaleax I was wondering about that.  If we don't have intl, do you know how typically convert to utf16 if intl is not enabled ?

Reply:
https://github.com/nodejs/node/pull/37954#issuecomment-1014514869 has some prior discussion on that (tl;dr: we don’t, we can use V8 in cases in which an Isolate is available but not here)

Reply:
Thanks for the pointer to the discussion, I can see that is trying to the code where I'd grabbed using icu for the conversion from :)

Reply:
I ended up making it so that the --without-intl reports an error if you try to configure it along with an externally shared js builtin and then guarding the code so that if --without-intl is used without the new options it is excluded.

My take is that it is acceptable for the new options to not be availble when --without-intl.  My reasoning is both that people using --without-intl is a small set of people at this point in time and that those that do are unlikely to be interested in externalizing builtins and that rolling our own icu convertion for this unlikely case is not worth the work/additional code in Node.js.  If that's wrong or we add native conversion helpers at some later time we can always do the work to remove that restriction.

Reply:
I also did confirm as Anna mentioned that an isolate is not available in the method where we do the coversions so using V8 is not possible in this case.